extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php

```diff
@@ -36,134 +38,171 @@ public static function onBeforePageDisplay() {
 
 	/**
 	 * Checks arguments for errors.
+	 * @param $modelData
 	 * @param array $args
-	 * @param string errorMessage Return parameter with the proper error message to show. Disregard if return is false
+	 * @internal param $errorMessage /Return parameter with the proper error message to show.
+	 * Disregard if return is false.
 	 * @return true if ok, false if error
 	 */
-	private function checkArguments( array $args, $modelData, &$errorMessage ) {
-		// mostrecent must be bool
-		if ( isset( $args['mostrecent'] ) &&
-			$args['mostrecent'] !== 'true' &&
-			$args['mostrecent'] !== 'false'
+	private function checkArguments( $modelData, array $args, &$errorMessage ) {
+		// PARAM_MOSTRECENT must be bool
+		if ( isset( $args[ static::PARAM_MOSTRECENT ] ) &&
+			$args[ static::PARAM_MOSTRECENT ] !== 'true' &&
+			$args[ static::PARAM_MOSTRECENT ] !== 'false'
 		) {
-			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', 'mostrecent' )->plain() .
-				wfMessage( 'embeddable-discussions-parameter-error-boolean' )->plain();
+			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error',
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "embeddable-discussions-parameter-error" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)